--- a/lib/GitLab/API/v3.pm	2016-01-08 14:58:11.769535426 +0000
+++ b/lib/GitLab/API/v3.pm	2016-01-08 15:02:42.371635104 +0000
@@ -232,8 +232,7 @@
     croak 'The #1 argument ($user_id) to user must be a scalar' if ref($_[0]) or (!defined $_[0]);
     my $path = sprintf('/users/%s', (map { uri_escape($_) } @_));
     $log->infof( 'Making %s request against %s with params %s.', 'GET', $path, undef );
-    $self->get( $path );
-    return;
+    return $self->get( $path );
 }
 
 =head2 create_user
@@ -253,8 +252,7 @@
     my $params = pop;
     my $path = sprintf('/users', (map { uri_escape($_) } @_));
     $log->infof( 'Making %s request against %s with params %s.', 'POST', $path, $params );
-    $self->post( $path, ( defined($params) ? $params : () ) );
-    return;
+    return $self->post( $path, ( defined($params) ? $params : () ) );
 }
 
 =head2 edit_user
@@ -276,8 +274,7 @@
     my $params = pop;
     my $path = sprintf('/users/%s', (map { uri_escape($_) } @_));
     $log->infof( 'Making %s request against %s with params %s.', 'PUT', $path, $params );
-    $self->put( $path, ( defined($params) ? $params : () ) );
-    return;
+    return $self->put( $path, ( defined($params) ? $params : () ) );
 }
 
 =head2 delete_user
@@ -386,8 +383,7 @@
     my $params = pop;
     my $path = sprintf('/user/keys', (map { uri_escape($_) } @_));
     $log->infof( 'Making %s request against %s with params %s.', 'POST', $path, $params );
-    $self->post( $path, ( defined($params) ? $params : () ) );
-    return;
+    return $self->post( $path, ( defined($params) ? $params : () ) );
 }
 
 =head2 create_user_ssh_key
@@ -409,8 +405,7 @@
     my $params = pop;
     my $path = sprintf('/users/%s/keys', (map { uri_escape($_) } @_));
     $log->infof( 'Making %s request against %s with params %s.', 'POST', $path, $params );
-    $self->post( $path, ( defined($params) ? $params : () ) );
-    return;
+    return $self->post( $path, ( defined($params) ? $params : () ) );
 }
 
 =head2 delete_current_user_ssh_key
@@ -429,8 +424,7 @@
     croak 'The #1 argument ($key_id) to delete_current_user_ssh_key must be a scalar' if ref($_[0]) or (!defined $_[0]);
     my $path = sprintf('/user/keys/%s', (map { uri_escape($_) } @_));
     $log->infof( 'Making %s request against %s with params %s.', 'DELETE', $path, undef );
-    $self->delete( $path );
-    return;
+    return $self->delete( $path );
 }
 
 =head2 delete_user_ssh_key
@@ -451,8 +445,7 @@
     croak 'The #2 argument ($key_id) to delete_user_ssh_key must be a scalar' if ref($_[1]) or (!defined $_[1]);
     my $path = sprintf('/users/%s/keys/%s', (map { uri_escape($_) } @_));
     $log->infof( 'Making %s request against %s with params %s.', 'DELETE', $path, undef );
-    $self->delete( $path );
-    return;
+    return $self->delete( $path );
 }
 
 =head1 SESSION METHODS
@@ -476,8 +469,7 @@
     my $params = pop;
     my $path = sprintf('/session', (map { uri_escape($_) } @_));
     $log->infof( 'Making %s request against %s with params %s.', 'POST', $path, $params );
-    $self->post( $path, ( defined($params) ? $params : () ) );
-    return;
+    return $self->post( $path, ( defined($params) ? $params : () ) );
 }
 
 =head1 PROJECT METHODS
@@ -613,8 +605,7 @@
     my $params = pop;
     my $path = sprintf('/projects/user/%s', (map { uri_escape($_) } @_));
     $log->infof( 'Making %s request against %s with params %s.', 'POST', $path, $params );
-    $self->post( $path, ( defined($params) ? $params : () ) );
-    return;
+    return $self->post( $path, ( defined($params) ? $params : () ) );
 }
 
 =head2 fork_project
@@ -633,8 +624,7 @@
     croak 'The #1 argument ($project_id) to fork_project must be a scalar' if ref($_[0]) or (!defined $_[0]);
     my $path = sprintf('/pojects/fork/%s', (map { uri_escape($_) } @_));
     $log->infof( 'Making %s request against %s with params %s.', 'POST', $path, undef );
-    $self->post( $path );
-    return;
+    return $self->post( $path );
 }
 
 =head2 delete_project
@@ -653,8 +643,7 @@
     croak 'The #1 argument ($project_id) to delete_project must be a scalar' if ref($_[0]) or (!defined $_[0]);
     my $path = sprintf('/projects/%s', (map { uri_escape($_) } @_));
     $log->infof( 'Making %s request against %s with params %s.', 'DELETE', $path, undef );
-    $self->delete( $path );
-    return;
+    return $self->delete( $path );
 }
 
 =head2 project_members
@@ -716,8 +705,7 @@
     my $params = pop;
     my $path = sprintf('/projects/%s/members', (map { uri_escape($_) } @_));
     $log->infof( 'Making %s request against %s with params %s.', 'POST', $path, $params );
-    $self->post( $path, ( defined($params) ? $params : () ) );
-    return;
+    return $self->post( $path, ( defined($params) ? $params : () ) );
 }
 
 =head2 edit_project_member
@@ -741,8 +729,7 @@
     my $params = pop;
     my $path = sprintf('/projects/%s/members/%s', (map { uri_escape($_) } @_));
     $log->infof( 'Making %s request against %s with params %s.', 'PUT', $path, $params );
-    $self->put( $path, ( defined($params) ? $params : () ) );
-    return;
+    return $self->put( $path, ( defined($params) ? $params : () ) );
 }
 
 =head2 remove_project_member
@@ -763,8 +750,7 @@
     croak 'The #2 argument ($user_id) to remove_project_member must be a scalar' if ref($_[1]) or (!defined $_[1]);
     my $path = sprintf('/projects/%s/members/%s', (map { uri_escape($_) } @_));
     $log->infof( 'Making %s request against %s with params %s.', 'DELETE', $path, undef );
-    $self->delete( $path );
-    return;
+    return $self->delete( $path );
 }
 
 =head2 project_hooks
@@ -826,8 +812,7 @@
     my $params = pop;
     my $path = sprintf('/projects/%s/hooks', (map { uri_escape($_) } @_));
     $log->infof( 'Making %s request against %s with params %s.', 'POST', $path, $params );
-    $self->post( $path, ( defined($params) ? $params : () ) );
-    return;
+    return $self->post( $path, ( defined($params) ? $params : () ) );
 }
 
 =head2 edit_project_hook
@@ -851,8 +836,7 @@
     my $params = pop;
     my $path = sprintf('/projects/%s/hooks/%s', (map { uri_escape($_) } @_));
     $log->infof( 'Making %s request against %s with params %s.', 'PUT', $path, $params );
-    $self->put( $path, ( defined($params) ? $params : () ) );
-    return;
+    return $self->put( $path, ( defined($params) ? $params : () ) );
 }
 
 =head2 delete_project_hook
@@ -894,8 +878,7 @@
     croak 'The #2 argument ($forked_from_id) to set_project_fork must be a scalar' if ref($_[1]) or (!defined $_[1]);
     my $path = sprintf('/projects/%s/fork/%s', (map { uri_escape($_) } @_));
     $log->infof( 'Making %s request against %s with params %s.', 'POST', $path, undef );
-    $self->post( $path );
-    return;
+    return $self->post( $path );
 }
 
 =head2 clear_project_fork
@@ -914,8 +897,7 @@
     croak 'The #1 argument ($project_id) to clear_project_fork must be a scalar' if ref($_[0]) or (!defined $_[0]);
     my $path = sprintf('/projects/%s/fork', (map { uri_escape($_) } @_));
     $log->infof( 'Making %s request against %s with params %s.', 'DELETE', $path, undef );
-    $self->delete( $path );
-    return;
+    return $self->delete( $path );
 }
 
 =head2 search_projects_by_name
@@ -1003,8 +985,7 @@
     my $params = pop;
     my $path = sprintf('/projects/%s/snippets', (map { uri_escape($_) } @_));
     $log->infof( 'Making %s request against %s with params %s.', 'POST', $path, $params );
-    $self->post( $path, ( defined($params) ? $params : () ) );
-    return;
+    return $self->post( $path, ( defined($params) ? $params : () ) );
 }
 
 =head2 edit_snippet
@@ -1028,8 +1009,7 @@
     my $params = pop;
     my $path = sprintf('/projects/%s/snippets/%s', (map { uri_escape($_) } @_));
     $log->infof( 'Making %s request against %s with params %s.', 'PUT', $path, $params );
-    $self->put( $path, ( defined($params) ? $params : () ) );
-    return;
+    return $self->put( $path, ( defined($params) ? $params : () ) );
 }
 
 =head2 delete_snippet
@@ -1050,8 +1030,7 @@
     croak 'The #2 argument ($snippet_id) to delete_snippet must be a scalar' if ref($_[1]) or (!defined $_[1]);
     my $path = sprintf('/projects/%s/snippets/%s', (map { uri_escape($_) } @_));
     $log->infof( 'Making %s request against %s with params %s.', 'DELETE', $path, undef );
-    $self->delete( $path );
-    return;
+    return $self->delete( $path );
 }
 
 =head2 raw_snippet
@@ -1117,8 +1096,7 @@
     my $params = pop;
     my $path = sprintf('/projects/%s/repository/tags', (map { uri_escape($_) } @_));
     $log->infof( 'Making %s request against %s with params %s.', 'POST', $path, $params );
-    $self->post( $path, ( defined($params) ? $params : () ) );
-    return;
+    return $self->post( $path, ( defined($params) ? $params : () ) );
 }
 
 =head2 tree
@@ -1296,8 +1274,7 @@
     my $params = pop;
     my $path = sprintf('/projects/%s/repository/files', (map { uri_escape($_) } @_));
     $log->infof( 'Making %s request against %s with params %s.', 'POST', $path, $params );
-    $self->post( $path, ( defined($params) ? $params : () ) );
-    return;
+    return $self->post( $path, ( defined($params) ? $params : () ) );
 }
 
 =head2 edit_file
@@ -1319,8 +1296,7 @@
     my $params = pop;
     my $path = sprintf('/projects/%s/repository/files', (map { uri_escape($_) } @_));
     $log->infof( 'Making %s request against %s with params %s.', 'PUT', $path, $params );
-    $self->put( $path, ( defined($params) ? $params : () ) );
-    return;
+    return $self->put( $path, ( defined($params) ? $params : () ) );
 }
 
 =head2 delete_file
@@ -1342,8 +1318,7 @@
     my $params = pop;
     my $path = sprintf('/projects/%s/repository/files', (map { uri_escape($_) } @_));
     $log->infof( 'Making %s request against %s with params %s.', 'DELETE', $path, $params );
-    $self->delete( $path, ( defined($params) ? $params : () ) );
-    return;
+    return $self->delete( $path, ( defined($params) ? $params : () ) );
 }
 
 =head1 COMMIT METHODS
@@ -1456,8 +1431,7 @@
     my $params = pop;
     my $path = sprintf('/projects/%s/repository/commits/%s/comments', (map { uri_escape($_) } @_));
     $log->infof( 'Making %s request against %s with params %s.', 'POST', $path, $params );
-    $self->post( $path, ( defined($params) ? $params : () ) );
-    return;
+    return $self->post( $path, ( defined($params) ? $params : () ) );
 }
 
 =head1 BRANCH METHODS
@@ -1522,8 +1496,7 @@
     croak 'The #2 argument ($branch_name) to protect_branch must be a scalar' if ref($_[1]) or (!defined $_[1]);
     my $path = sprintf('/projects/%s/repository/branches/%s/protect', (map { uri_escape($_) } @_));
     $log->infof( 'Making %s request against %s with params %s.', 'PUT', $path, undef );
-    $self->put( $path );
-    return;
+    return $self->put( $path );
 }
 
 =head2 unprotect_branch
@@ -1544,8 +1517,7 @@
     croak 'The #2 argument ($branch_name) to unprotect_branch must be a scalar' if ref($_[1]) or (!defined $_[1]);
     my $path = sprintf('/projects/%s/repository/branches/%s/unprotect', (map { uri_escape($_) } @_));
     $log->infof( 'Making %s request against %s with params %s.', 'PUT', $path, undef );
-    $self->put( $path );
-    return;
+    return $self->put( $path );
 }
 
 =head2 create_branch
@@ -1588,8 +1560,7 @@
     croak 'The #2 argument ($branch_name) to delete_branch must be a scalar' if ref($_[1]) or (!defined $_[1]);
     my $path = sprintf('/projects/%s/repository/branches/%s', (map { uri_escape($_) } @_));
     $log->infof( 'Making %s request against %s with params %s.', 'DELETE', $path, undef );
-    $self->delete( $path );
-    return;
+    return $self->delete( $path );
 }
 
 =head1 MERGE REQUEST METHODS
@@ -1703,8 +1674,7 @@
     my $params = pop;
     my $path = sprintf('/projects/%s/merge_requests/%s/merge', (map { uri_escape($_) } @_));
     $log->infof( 'Making %s request against %s with params %s.', 'PUT', $path, $params );
-    $self->put( $path, ( defined($params) ? $params : () ) );
-    return;
+    return $self->put( $path, ( defined($params) ? $params : () ) );
 }
 
 =head2 add_merge_request_comment
@@ -1728,8 +1698,7 @@
     my $params = pop;
     my $path = sprintf('/projects/%s/merge_requests/%s/comments', (map { uri_escape($_) } @_));
     $log->infof( 'Making %s request against %s with params %s.', 'POST', $path, $params );
-    $self->post( $path, ( defined($params) ? $params : () ) );
-    return;
+    return $self->post( $path, ( defined($params) ? $params : () ) );
 }
 
 =head2 merge_request_comments
@@ -1930,8 +1899,7 @@
     my $params = pop;
     my $path = sprintf('/projects/%s/labels', (map { uri_escape($_) } @_));
     $log->infof( 'Making %s request against %s with params %s.', 'DELETE', $path, $params );
-    $self->delete( $path, ( defined($params) ? $params : () ) );
-    return;
+    return $self->delete( $path, ( defined($params) ? $params : () ) );
 }
 
 =head2 edit_label
@@ -2019,8 +1987,7 @@
     my $params = pop;
     my $path = sprintf('/projects/%s/milestones', (map { uri_escape($_) } @_));
     $log->infof( 'Making %s request against %s with params %s.', 'POST', $path, $params );
-    $self->post( $path, ( defined($params) ? $params : () ) );
-    return;
+    return $self->post( $path, ( defined($params) ? $params : () ) );
 }
 
 =head2 edit_milestone
@@ -2044,8 +2011,7 @@
     my $params = pop;
     my $path = sprintf('/projects/%s/milestones/%s', (map { uri_escape($_) } @_));
     $log->infof( 'Making %s request against %s with params %s.', 'PUT', $path, $params );
-    $self->put( $path, ( defined($params) ? $params : () ) );
-    return;
+    return $self->put( $path, ( defined($params) ? $params : () ) );
 }
 
 =head1 NOTE METHODS
@@ -2123,8 +2089,7 @@
     my $params = pop;
     my $path = sprintf('/projects/%s/%s/%s/notes', (map { uri_escape($_) } @_));
     $log->infof( 'Making %s request against %s with params %s.', 'POST', $path, $params );
-    $self->post( $path, ( defined($params) ? $params : () ) );
-    return;
+    return $self->post( $path, ( defined($params) ? $params : () ) );
 }
 
 =head1 DEPLOY KEY METHODS
@@ -2190,8 +2155,7 @@
     my $params = pop;
     my $path = sprintf('/projects/%s/keys', (map { uri_escape($_) } @_));
     $log->infof( 'Making %s request against %s with params %s.', 'POST', $path, $params );
-    $self->post( $path, ( defined($params) ? $params : () ) );
-    return;
+    return $self->post( $path, ( defined($params) ? $params : () ) );
 }
 
 =head2 delete_deploy_key
@@ -2212,8 +2176,7 @@
     croak 'The #2 argument ($key_id) to delete_deploy_key must be a scalar' if ref($_[1]) or (!defined $_[1]);
     my $path = sprintf('/projects/%s/keys/%s', (map { uri_escape($_) } @_));
     $log->infof( 'Making %s request against %s with params %s.', 'DELETE', $path, undef );
-    $self->delete( $path );
-    return;
+    return $self->delete( $path );
 }
 
 =head1 SYSTEM HOOK METHODS
@@ -2253,8 +2216,7 @@
     my $params = pop;
     my $path = sprintf('/hooks', (map { uri_escape($_) } @_));
     $log->infof( 'Making %s request against %s with params %s.', 'POST', $path, $params );
-    $self->post( $path, ( defined($params) ? $params : () ) );
-    return;
+    return $self->post( $path, ( defined($params) ? $params : () ) );
 }
 
 =head2 test_hook
@@ -2292,8 +2254,7 @@
     croak 'The #1 argument ($hook_id) to delete_hook must be a scalar' if ref($_[0]) or (!defined $_[0]);
     my $path = sprintf('/hooks/%s', (map { uri_escape($_) } @_));
     $log->infof( 'Making %s request against %s with params %s.', 'DELETE', $path, undef );
-    $self->delete( $path );
-    return;
+    return $self->delete( $path );
 }
 
 =head1 GROUP METHODS
@@ -2352,8 +2313,7 @@
     my $params = pop;
     my $path = sprintf('/groups', (map { uri_escape($_) } @_));
     $log->infof( 'Making %s request against %s with params %s.', 'POST', $path, $params );
-    $self->post( $path, ( defined($params) ? $params : () ) );
-    return;
+    return $self->post( $path, ( defined($params) ? $params : () ) );
 }
 
 =head2 transfer_project
@@ -2374,8 +2334,7 @@
     croak 'The #2 argument ($project_id) to transfer_project must be a scalar' if ref($_[1]) or (!defined $_[1]);
     my $path = sprintf('/groups/%s/projects/%s', (map { uri_escape($_) } @_));
     $log->infof( 'Making %s request against %s with params %s.', 'POST', $path, undef );
-    $self->post( $path );
-    return;
+    return $self->post( $path );
 }
 
 =head2 delete_group
@@ -2394,8 +2353,7 @@
     croak 'The #1 argument ($group_id) to delete_group must be a scalar' if ref($_[0]) or (!defined $_[0]);
     my $path = sprintf('/groups/%s', (map { uri_escape($_) } @_));
     $log->infof( 'Making %s request against %s with params %s.', 'DELETE', $path, undef );
-    $self->delete( $path );
-    return;
+    return $self->delete( $path );
 }
 
 =head2 group_members
@@ -2436,8 +2394,7 @@
     my $params = pop;
     my $path = sprintf('/groups/%s/members', (map { uri_escape($_) } @_));
     $log->infof( 'Making %s request against %s with params %s.', 'POST', $path, $params );
-    $self->post( $path, ( defined($params) ? $params : () ) );
-    return;
+    return $self->post( $path, ( defined($params) ? $params : () ) );
 }
 
 =head2 remove_group_member
@@ -2458,8 +2415,7 @@
     croak 'The #2 argument ($user_id) to remove_group_member must be a scalar' if ref($_[1]) or (!defined $_[1]);
     my $path = sprintf('/groups/%s/members/%s', (map { uri_escape($_) } @_));
     $log->infof( 'Making %s request against %s with params %s.', 'DELETE', $path, undef );
-    $self->delete( $path );
-    return;
+    return $self->delete( $path );
 }
 
 =head1 SERVICE METHODS
@@ -2487,8 +2443,7 @@
     my $params = pop;
     my $path = sprintf('/projects/%s/services/%s', (map { uri_escape($_) } @_));
     $log->infof( 'Making %s request against %s with params %s.', 'PUT', $path, $params );
-    $self->put( $path, ( defined($params) ? $params : () ) );
-    return;
+    return $self->put( $path, ( defined($params) ? $params : () ) );
 }
 
 =head2 delete_project_service
@@ -2509,8 +2464,7 @@
     croak 'The #2 argument ($service_name) to delete_project_service must be a scalar' if ref($_[1]) or (!defined $_[1]);
     my $path = sprintf('/projects/%s/services/%s', (map { uri_escape($_) } @_));
     $log->infof( 'Making %s request against %s with params %s.', 'DELETE', $path, undef );
-    $self->delete( $path );
-    return;
+    return $self->delete( $path );
 }
 
 1;
