--- ./src/plugins/slurm/slurm-spank-auks.c	2021-02-24 09:12:53.780172640 +0300
+++ ./src/plugins/slurm/slurm-spank-auks.c2	2021-02-24 10:07:30.857002178 +0300
@@ -287,12 +287,30 @@
 		return -1;
 		break;
 	}
+	
+	uid_t uid = geteuid();
+	gid_t gid = getegid();
 
 	renewer_pid = fork();
 	if ( renewer_pid == -1 ) {
 		xerror("unable to launch renewer process");
 	}
 	else if ( renewer_pid == 0 ) {
+		seteuid(getuid());
+		setegid(getgid());
+
+		if ( setgid(gid) ) {
+			xerror("unable to switch to user gid : %s",
+			       strerror(errno));
+			exit(1);
+		}
+
+		if ( setuid(uid) ) {
+			xerror("unable to switch to user uid : %s",
+			       strerror(errno));
+			exit(1);
+		}
+
 		sigset_t mask;
 		sigemptyset(&mask);
 		sigprocmask(SIG_SETMASK, &mask, NULL);
