--- a/src/plugins/slurm/slurm-spank-auks.c	2020-12-17 07:34:45.000000000 +0000
+++ b/src/plugins/slurm/slurm-spank-auks.c	2020-12-17 07:37:01.000000000 +0000
@@ -297,8 +297,9 @@
 		sigemptyset(&mask);
 		sigprocmask(SIG_SETMASK, &mask, NULL);
 		char *argv[4];
-		argv[0]= BINDIR "/auks" ;
-		argv[1]="-R";argv[2]="loop";
+		argv[0]= "/bin/bash";
+		argv[1]="-c";
+		argv[2]= "exec " BINDIR "/auks -R loop";
 		argv[3]=NULL;
 		setenv("KRB5CCNAME",auks_credcache,1);
 		chdir("/");
