--- a/src/plugins/slurm/slurm-spank-auks.c	2021-02-24 09:12:53.780172640 +0300
+++ b/src/plugins/slurm/slurm-spank-auks.c	2021-02-24 09:24:34.190207449 +0300
@@ -287,12 +287,37 @@
 		return -1;
 		break;
 	}
+	
+	uid_t uid;
+	gid_t gid;
+
+	/* get slurm job user uid */
+	if (spank_get_item (sp, S_JOB_UID, &uid) != ESPANK_SUCCESS) {
+		xerror("failed to get uid: %s", strerror(errno));
+		return (-1);
+	}
+	if (spank_get_item (sp, S_JOB_GID, &gid) != ESPANK_SUCCESS) {
+		xerror("failed to get gid: %s", strerror(errno));
+		return (-1);
+	}
 
 	renewer_pid = fork();
 	if ( renewer_pid == -1 ) {
 		xerror("unable to launch renewer process");
 	}
 	else if ( renewer_pid == 0 ) {
+		if ( setgid(gid) ) {
+			xerror("unable to switch to user gid : %s",
+			       strerror(errno));
+			exit(1);
+		}
+
+		if ( setuid(uid) ) {
+			xerror("unable to switch to user uid : %s",
+			       strerror(errno));
+			exit(1);
+		}
+
 		sigset_t mask;
 		sigemptyset(&mask);
 		sigprocmask(SIG_SETMASK, &mask, NULL);
