--- bin/dig/dighost.c2	2018-03-08 23:56:40.000000000 +0300
+++ bin/dig/dighost.c	2018-06-25 17:22:24.682179603 +0300
@@ -409,10 +409,6 @@
 		exitcode = 10;
 	if (fatalexit != 0)
 		exitcode = fatalexit;
-	if (dighost_pre_exit_hook != NULL) {
-		dighost_pre_exit_hook();
-	}
-	exit(exitcode);
 }
 
 void
@@ -548,9 +544,11 @@
 
 	result = bind9_getaddresses(opt, 0, sockaddrs,
 				    DIG_MAX_ADDRESSES, &count);
-	if (result != ISC_R_SUCCESS)
+	if (result != ISC_R_SUCCESS){
 		fatal("couldn't get address for '%s': %s",
 		      opt, isc_result_totext(result));
+		return;
+	}
 
 	flush_server_list();
 
@@ -847,8 +845,10 @@
 	debug("requeue_lookup()");
 
 	lookup_counter++;
-	if (lookup_counter > LOOKUP_LIMIT)
+	if (lookup_counter > LOOKUP_LIMIT){
 		fatal("too many lookups");
+		return NULL;
+	}
 
 	looknew = clone_lookup(lookold, servers);
 	INSIST(looknew != NULL);
@@ -1838,6 +1838,7 @@
 	if (numLookups == 0 && addresses_result != ISC_R_SUCCESS) {
 		fatal("couldn't get address for '%s': %s",
 		      bad_namestr, isc_result_totext(result));
+		return 0;
 	}
 
 	if (lookup == NULL &&
@@ -4029,6 +4030,7 @@
 
 	result = bind9_getaddresses(host, 0, sockaddrs,
 				    DIG_MAX_ADDRESSES, &count);
+	result = ISC_R_SUCCESS;
 	if (resultp != NULL)
 		*resultp = result;
 	if (result != ISC_R_SUCCESS) {
