--- a/bin/dig/dig.c	2013-06-29 16:45:21.626605921 +0400
+++ b/bin/dig/dig.c	2013-06-29 19:36:09.557242247 +0400
@@ -61,6 +61,7 @@
 static FILE *batchfp = NULL;
 static char *argv0;
 static int addresscount = 0;
+static int prompt = 0;
 
 static char domainopt[DNS_NAME_MAXTEXT];
 
@@ -1551,7 +1551,7 @@
 /*%
  * #true returned if value was used
  */
-static const char *single_dash_opts = "46dhimnruv";
+static const char *single_dash_opts = "46dhimnruvs";
 static const char *dash_opts = "46bcdfhikmnpqrtvyx";
 static bool
 dash_option(char *option, char *next, dig_lookup_t **lookup,
@@ -1247,6 +1249,9 @@
 			help();
 			exit(0);
 			break;
+		case 's':
+			prompt = 1;
+			break;
 		case 'i':
 			ip6_int = ISC_TRUE;
 			break;
@@ -1717,6 +1722,7 @@
 		}
 		/* XXX Remove code dup from shutdown code */
 	next_line:
+		if((prompt == 1) && (batchname != NULL)){ printf("dig> "); }
 		if (fgets(batchline, sizeof(batchline), batchfp) != 0) {
 			bargc = 1;
 			debug("batch line %s", batchline);
@@ -1790,6 +1796,7 @@
 		return;
 	}
 
+	if((prompt == 1) && (batchname != NULL)){ printf("dig> "); }
 	if (fgets(batchline, sizeof(batchline), batchfp) != 0) {
 		debug("batch line %s", batchline);
 		bargc = 1;
