--- a/lib/File/Stamped.pm	2012-12-19 14:35:25.054483619 +0400
+++ a/lib/File/Stamped.pm	2012-12-19 15:27:48.979678830 +0400
@@ -7,6 +7,7 @@
 use POSIX ();
 use SelectSaver ();
 use File::Path qw(mkpath);
+use IO::Compress::Gzip;
 
 sub new {
     my $class = shift;
@@ -63,6 +64,9 @@
             my $saver = SelectSaver->new($fh);
             $|=1;
         }
+        if(*$self->{gzip}){
+                $fh = new IO::Compress::Gzip $fh;
+        }
     }
     print {$fh} @_
         or die "Cannot write to $fname: $!";
