--- a/lib/File/Stamped.pm	2012-12-19 14:33:27.694476407 +0400
+++ b/lib/File/Stamped.pm	2012-12-19 14:35:25.054483619 +0400
@@ -6,6 +6,7 @@
 use Carp ();
 use POSIX ();
 use SelectSaver ();
+use File::Path qw(mkpath);
 
 sub new {
     my $class = shift;
@@ -54,7 +55,10 @@
         }
     }
     unless ($fh) {
-        open $fh, *$self->{iomode}, $fname or die "Cannot open file($fname): $!";
+        my ($dir) = ($fname =~ /^(.*\/)[^\/]+$/);
+	mkpath $dir if $dir;
+	
+	open $fh, *$self->{iomode}, $fname or die "Cannot open file($fname): $!";
         if (*$self->{autoflush}) {
             my $saver = SelectSaver->new($fh);
             $|=1;
